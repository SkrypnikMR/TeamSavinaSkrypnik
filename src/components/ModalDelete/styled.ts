import styled from 'styled-components';

export const StModalDelete = styled.div`
    p{
        text-align: center;
    }
`;
