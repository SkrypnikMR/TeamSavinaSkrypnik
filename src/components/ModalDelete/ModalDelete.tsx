import React from 'react';
import { useTranslation } from 'react-i18next';
import Button from '../UI/Button';
import { StModalButtonBox } from '../ModalCreateRoom/styled';
import { StModalDelete } from './styled';

const ModalDelete = ({ handlecloseModal }) => {
    const { t } = useTranslation();
    return (
        <StModalDelete>
            <p>{t('are_you_sure')}</p>
            <StModalButtonBox>
                <Button 
                    content="OK"
                    focusColor="black"
                    onClick={handlecloseModal}
                    id="okButton"
                    name="okButton"
                    value="OK"
                    type="button"
                    title="OK"
                    bgColorShadow="0 14px 28px rgba(0, 0, 0, 0.6), 0 10px 10px rgba(0, 0, 0, 0.22)"
                />
                <Button
                    content={t('cancel')}
                    focusColor="black"
                    onClick={handlecloseModal}
                    id="CancelButton"
                    name="CancelButton"
                    value="Cancel"
                    type="button"
                    title="Cancel"
                    bgColorShadow="0 14px 28px rgba(0, 0, 0, 0.6), 0 10px 10px rgba(0, 0, 0, 0.22)"
                />
            </StModalButtonBox>
        </StModalDelete>
    );
};

export default ModalDelete;
