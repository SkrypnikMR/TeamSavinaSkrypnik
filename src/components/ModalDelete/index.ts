import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import ModalDelete from './ModalDelete';
import { exitGame } from '../../store/game/actions';

const mapDispatchToProps = (dispatch: Dispatch) => ({ exitGame: () => dispatch(exitGame()) });

export default connect(null, mapDispatchToProps)(ModalDelete);
