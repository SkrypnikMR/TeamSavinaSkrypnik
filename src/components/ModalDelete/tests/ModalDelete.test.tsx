import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalDelete from '../ModalDelete';

describe('ModalCreateRoom', () => {
    let props;
    let modal;
    beforeEach(() => {
        modal = document.createElement('div');
        modal.setAttribute('id', 'modal');
        document.body.append(modal);
    props = {
        handlecloseModal: jest.fn(),
        exitGame: jest.fn(),
    };
    });
    it('should snapshot', () => {
        const component = shallow(<ModalDelete {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StModalButtonBox', () => {
        const component = mount(<ModalDelete {...props}/>);
        expect(component.find('styled__StModalButtonBox')).toHaveLength(1);
    });
    it('should click on first button, then call exitGame and handlecloseModal', () => {
        const component = mount(<ModalDelete {...props}/>);
        component.find('Button').at(0).props().onClick();
        expect(props.exitGame).toHaveBeenCalled();
        expect(props.handlecloseModal).toHaveBeenCalled();
    });
});
