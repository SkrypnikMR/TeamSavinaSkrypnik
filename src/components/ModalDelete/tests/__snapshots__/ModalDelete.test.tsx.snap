// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCreateRoom should snapshot 1`] = `"<div class=\\"styled__StModalDelete-sc-1bs2vgy-0 bJVuRx\\"><p>are_you_sure</p><div class=\\"styled__StModalButtonBox-sc-1lbp0ln-0 kgpYEL\\"><button id=\\"okButton\\" name=\\"okButton\\" type=\\"button\\" value=\\"OK\\" title=\\"OK\\" class=\\"styled__StButton-sc-1y35zax-0 hBgoPh\\">OK</button><button id=\\"CancelButton\\" name=\\"CancelButton\\" type=\\"button\\" value=\\"Cancel\\" title=\\"Cancel\\" class=\\"styled__StButton-sc-1y35zax-0 efYwVL\\">cancel</button></div></div>"`;
